compair/kaltura/kaltura.py

```diff
@@ -70,4 +74,30 @@ def complete_upload_for_token(cls, upload_token_id):
         KalturaSession._api_end(kaltura_media.upload_ks)
         db.session.commit()
 
-        return kaltura_media
\ No newline at end of file
+        return kaltura_media
+
+    @classmethod
+    def get_direct_access_url(cls, entry_id, download_url=None, expiry=300):
+        """
+            Generate a new Kaltura Session (KS) and return an URL for direct download of Kaltura media file.
+            :param entry_id: the Kaltura entry ID of the file to download
+            :param download_url: optional URL to download the file. If not provided, an URL will be generated to download the file via playManifest API
+            :param expiry: optional expiry time of the KS, in seconds. Default is 300 seconds.
+            :return: a URL with KS attached for downloading the given Kaltura media file
+        """
+        # Use system id instead of user's own id to create the session,
+        # even if KALTURA_USE_GLOBAL_UNIQUE_IDENTIFIER is defined.
+        kaltura_user_id = KalturaCore.user_id()
+
+        url = download_url
+        if not url:
```

Comment:
if there is no download_url then the file hasn't be uploaded yet. It would be most likely be better to error out then to try and create one ourselves